addon/store.js

```diff
@@ -131,13 +123,41 @@ var Store = Ember.Object.extend({
 
             var attr = params[0];
             var value = options[attr];
-            var func = function(m) {
+
+            return this._findWithFilterFunc(type, function filterFunction(m) {
                 return get(m, attr) === value;
-            };
-            return this._findWithFilterFunc(type, func);
+            });
         }
+
         return this._findByIdComputed(type, options);
     },
+    findOne(type) {
+        var store = this;
+
+        return Ember.ObjectProxy.extend({
```

Comment:
For now I think the ObjectProxy refactor is out of scope for v4.0.0 (mainly because I had a client sponsor me personally doing the upgrade and this would be another breaking change). That said, I do hate that it's here for all the reasons you mention and it's been a nifty work to my core problem where I return objects right away from the routes model hook (never waiting for a promise). It sounds like our next big release could look at swapping this for the model + content that you suggested or something else that's a better fit (I looked at generators for this a while back but wanted to see ember-data adopt that in place of ObjectProxy first to ensure people liked using it).

Cool if we table this for another day?